File: enode/src/main/java/org/enodeframework/commanding/ICommandExecuteContext.java
package org.enodeframework.commanding;

/**
 * Represents a context environment for command executor executing command.
 */
public interface ICommandExecuteContext extends ICommandContext, ITrackingContext {
    /**
     * Notify the given command is executed.
     */
    void onCommandExecutedAsync(CommandResult commandResult);
}

File: enode/src/main/java/org/enodeframework/commanding/ICommandExecuteContext.kt
package org.enodeframework.commanding

import java.util.concurrent.CompletableFuture

/**
 * Represents a context environment for command executor executing command.
 */
interface ICommandExecuteContext : ICommandContext, ITrackingContext {
    /**
     * Notify the given command is executed.
     */
    fun onCommandExecutedAsync(commandResult: CommandResult): CompletableFuture<Void>
}